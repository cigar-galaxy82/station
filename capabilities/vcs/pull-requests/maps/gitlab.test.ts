import { SuperfaceTest } from '@superfaceai/testing-lib';

describe(`vcs/pull-requests/gitlab`, () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = new SuperfaceTest();
  });

  describe('PullRequests', () => {
    it('should perform successfully', async () => {
      await expect(
        superface.run({
          profile: 'vcs/pull-requests',
          provider: 'gitlab',
          useCase: 'PullRequests',
          input: {
            owner: 'Jakub-Vacek',
            repo: 'empty-test',
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });
  it('should handle error', async () => {
    await expect(
      superface.run({
        profile: 'vcs/pull-requests',
        provider: 'gitlab',
        useCase: 'PullRequests',
        input: {
          owner: 'Jakub-Vacek',
          repo: 'made-up',
        },
      })
    ).resolves.toMatchSnapshot();
  }, 10000);
});
