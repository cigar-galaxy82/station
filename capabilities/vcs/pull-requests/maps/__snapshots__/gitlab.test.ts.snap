// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vcs/pull-requests/gitlab PullRequests should handle error 1`] = `
Err {
  "error": "MappedError: Expected error
Properites: {
  \\"message\\": \\"404 Project Not Found\\",
  \\"statusCode\\": 404
}
AST Path: definitions[0].statements[0].statements[0]
Original Map Location: Line 12, column 5",
}
`;

exports[`vcs/pull-requests/gitlab PullRequests should perform successfully - multiple pages of PRs 1`] = `
Ok {
  "value": Object {
    "pullRequests": Array [
      Object {
        "id": 21,
        "sha": "4f9b3d7fcb0eaa5e07b58ecb6e70a1673f90af48",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/21",
      },
      Object {
        "id": 20,
        "sha": "42d674a58bfc38be4a37c431b73d14ef22a2f7d1",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/20",
      },
      Object {
        "id": 19,
        "sha": "69fec858a482fedca2d4d4122349e30557417d6b",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/19",
      },
      Object {
        "id": 18,
        "sha": "c837315c9669ef65ff3283937d916103f9b90a90",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/18",
      },
      Object {
        "id": 17,
        "sha": "7a0acd873a4bc24c4ed4b168ae1889d15d325735",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/17",
      },
      Object {
        "id": 16,
        "sha": "72508f3557307db7310699a6eca79fa7e5fb8df1",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/16",
      },
      Object {
        "id": 15,
        "sha": "38aa6ef7bab4634a83a1546c629df9f13ce6ba2a",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/15",
      },
      Object {
        "id": 14,
        "sha": "0144277897217a7e0409f1703350ef964ab7b5f7",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/14",
      },
      Object {
        "id": 13,
        "sha": "c614018ad551e00fff2f03a5dc87b3605cb34f3a",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/13",
      },
      Object {
        "id": 12,
        "sha": "a4da426b4793e97ed1559e36e96948be7019a79a",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/12",
      },
      Object {
        "id": 11,
        "sha": "383393ecd14c5036732bde9b1b15713cbf1fcc0e",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/11",
      },
      Object {
        "id": 10,
        "sha": "5345812884b1e6a05696d5ad67fc96c393108406",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/10",
      },
      Object {
        "id": 9,
        "sha": "acc5052126adb46174a4a1e2ffa36a20ed4755a9",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/9",
      },
      Object {
        "id": 8,
        "sha": "edea53c6ef10c69d2c4f390d741b2de05224ee63",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/8",
      },
      Object {
        "id": 7,
        "sha": "22d5d8395c7e30a2426262ff8f99e02bd6de3b69",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/7",
      },
      Object {
        "id": 6,
        "sha": "e7c86b15bf301e9bc05a5f1c22cc1b04918ec137",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/6",
      },
      Object {
        "id": 5,
        "sha": "9f31e6b738043e0cea30b91ad5983847fe56e181",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/5",
      },
      Object {
        "id": 4,
        "sha": "653fd4178b75d8fa5b2f5ab4150c4d19723895fe",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/4",
      },
      Object {
        "id": 3,
        "sha": "1182ed03c3950ba8cca301a5002226b0f7ccfd9d",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/3",
      },
      Object {
        "id": 2,
        "sha": "dfe6d68d7d0f7da5c7d5d9d01cbfe5bc3a76dfb4",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/2",
      },
      Object {
        "id": 1,
        "sha": "8c64ce23d626c5bf345ce90fa5af329569d62c9a",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/empty-test/-/merge_requests/1",
      },
    ],
    "statusCode": 200,
  },
}
`;

exports[`vcs/pull-requests/gitlab PullRequests should perform successfully - only one page of PRs 1`] = `
Ok {
  "value": Object {
    "pullRequests": Array [
      Object {
        "id": 1,
        "sha": "ece012d5c06d592cd231ef077d7bf8108fbd7808",
        "title": "Update README.md",
        "url": "https://gitlab.com/Jakub-Vacek/single-pr/-/merge_requests/1",
      },
    ],
    "statusCode": 200,
  },
}
`;
